pkg/netceptor/tlsconfig.go

```diff
@@ -39,14 +41,39 @@ func decodeFingerprints(fingerprints []string) ([][]byte, error) {
 	return fingerprintBytes, nil
 }
 
+func checkCertificatesMatchNodeID(certbytes []byte, n *Netceptor, certName string, certPath string) error {
+	block, _ := pem.Decode(certbytes)
+
+	if block == nil {
+		return fmt.Errorf("failed to parse certfifcate PEM")
+	}
+
+	parsedCert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return err
+	}
+
+	found, receptorNames, err := utils.ParseReceptorNamesFromCert(parsedCert, n.nodeID)
+	if err != nil {
+		return err
+	}
+
+	if !found {
```

Comment:
we still need to return without error here if len(receptorNames) == 0

if I create a cert without a nodeID, and use that on a config with node ID foo, but I see this error

```
[sbf@fedora sockceptor]$ ./receptor -c foo.yml
Error: MainInstance.nodeID=foo not found in certificate name(s); names found=[]; cfg section=server2; server cert=/home/sbf/sockceptor/certs/bob.crt
```

we want this to be permissible

Reply:
I'm okay with that option